File: crime-analytics-backend/src/main/java/com/crimeanalytics/service/OfficerService.java
package com.crimeanalytics.service;

import com.crimeanalytics.dto.OfficerRequestDTO;
import com.crimeanalytics.dto.OfficerResponseDTO;

import java.util.List;

public interface OfficerService {

    OfficerResponseDTO createOfficer(OfficerRequestDTO requestDTO);

    OfficerResponseDTO getOfficerById(Long id);

    List<OfficerResponseDTO> getAllOfficers();

    OfficerResponseDTO updateOfficer(Long id, OfficerRequestDTO requestDTO);

    void deleteOfficer(Long id);
}

File: crime-analytics-frontend/src/services/officerService.js
import axiosInstance from '../api/axiosInstance';

export const officerService = {
  async getAllOfficers() {
    const response = await axiosInstance.get('/officers');
    return response.data;
  },

  async getOfficerById(id) {
    const response = await axiosInstance.get(`/officers/${id}`);
    return response.data;
  },

  async createOfficer(data) {
    // data: { badgeNumber, name, department, rank, phoneNumber, email }
    const response = await axiosInstance.post('/officers', data);
    return response.data;
  },

  async updateOfficer(id, data) {
    const response = await axiosInstance.put(`/officers/${id}`, data);
    return response.data;
  },

  async deleteOfficer(id) {
    const response = await axiosInstance.delete(`/officers/${id}`);
    return response.data;
  }
};
